### ISSUE TITLE
Allow removing credit card info.

### ISSUE THREAD

**Issue by serefarikan** [docker/hub-feedback#2128]
## Problem description
Once a credit card is saved as a payment method, it is impossible to remove the card.

This issue was reported a number of times and yet the issues are closed with no satisfactory solution. This is a feature directly associated to paying customers (who register their cards to pay) so please do not keep closing issues related to this problem and instead fix it. 

https://github.com/docker/hub-feedback/issues/1596
https://github.com/docker/hub-feedback/issues/861

**Comment by ankudinov** [comment]
+1
Keep in mind: lack of this feature may have legal consequences in some countries. This must be removable without engaging support.
I'd also suggest adding PayPal support.

**Comment by jaycoady** [comment]
I am also having issues with this non feature. We are not allowed to store cards on file per government rules. Please make a solution.

**Comment by ivan-ottinger** [comment]
+1 Please add the possibility to remove payment card.

It would be also great to provide a tool to deactivate auto-renewal without having to downgrade the plan.

**Comment by pola-tech** [comment]
Same +1

**Comment by st333z** [comment]
remove the fucking credit card details from my account your pricks!

**Comment by pm-pm** [comment]
My card was blocked after sponsoring,  because you do not request the security items (PSD2) in payment process.
NOT good!

And i can not delete/remove my card from github. 

**I would never have made a payment if i had known that the card details would remain stored!**

Remove the Carddata! Fast!

**Comment by leslielylee** [comment]
Hi @serefarikan / @ankudinov / @jaycoady / @ivan-ottinger / @joehays / @acosenti / @pola-tech,

For recurring subscriptions, we need at least one card on file for a paid plan. However, this is something that we'd like to look into and investigate further based on the feedback we've read here. If you're willing to chat with us about it, please reach out at leslie.lee@docker.com, and I'm happy to schedule a time!

**Comment by michelekorell** [comment]
Hello, under GDPR, data controllers and processors are obliged to return or delete all personal data after the end of services, or on expiry of a contract or agreement, unless it’s necessary to retain the data by law. Can you ensure to me that if i deactivate my account you delete every data about my billing information and credit cards?

**Comment by likanzhan** [comment]
Does this problem has a solution yet?

**Comment by fcavassini-rsc** [comment]
This is abusive

**Comment by DonCorleone92** [comment]
Hi All,
Please give an option to delete the credit card information. This is highly unethical and corrupt practice.

**Comment by serefarikan** [comment]
@leslielylee thanks for responding but please note that as a subscriber I should have the right to remove the card info. It is up to docker to decide what to do to my subscription if there's no card, but not allowing me to remove the card is not a good user experience. Freeze the subscription if there's no card on record, cancel it, etc etc. Pretty much every major subscription service allows its users to manage their payment settings, but you're not allowing this.

**Comment by ro-explorer** [comment]
I ended up creating a burner card with Revolut, I updated the payment info then I destroyed the card. 
Special situations requires special solutions.

**Comment by scrdcow** [comment]
I want to remove my card and I have no recurring subscription.

**Comment by anarchy-79** [comment]
Ok it seems this is not happening, so I guess it's time to take legal action. Thank god for Swedish consumer protection laws. Also this type of scummy behavior instantly blacklists MS Github for me, personally, I will not use any product that tries to play sneaky with my CC info. What a disgrace.

**Comment by MartinRaSt** [comment]
I want to remove my credit card,  I don't want to share it with github. In case of necessary payment, I am able to fill it.

**Comment by supset** [comment]
Still waiting

**Comment by supset** [comment]
> Hi @serefarikan / @ankudinov / @jaycoady / @ivan-ottinger / @joehays / @acosenti / @pola-tech,
> 
> For recurring subscriptions, we need at least one card on file for a paid plan. However, this is something that we'd like to look into and investigate further based on the feedback we've read here. If you're willing to chat with us about it, please reach out at [leslie.lee@docker.com](mailto:leslie.lee@docker.com), and I'm happy to schedule a time!

Email is bogus invalid

**Comment by mattdeluco** [comment]
Hello everyone,

I manage the Billing Engineering Team here at Docker and can be your PoC on this issue. The email `leslie.lee@docker.com` has been deactivated.

We hear and understand your frustration on this issue and among our many endeavours do ultimately intend to give you the necessary tools to better manage your account.

It is certainly not in Docker's values to deceive anyone: a subscription implies periodic, recurring payments. In order to facilitate that, we must have a means of payment on file with our payments provider.

While this was the thought process behind our initial design, with your feedback we recognize there are situations in which our users wish not to, or may not, have their CC data stored permanently, and are willing to manually renew their subscription.

To that end, I am currently investigating short-term and long-term solutions to this issue. Expect further updates here as I have more answers.

**Comment by bs-git** [comment]
Yes, make this happen, this is barely legal and next step for me would be delete my account and use a different git host....

**Comment by serefarikan** [comment]
Thanks @mattdeluco for the clarification. I'm the one who originally created this issue. To rephrase this as simply as possible, as a subscriber, I just want my subscription to work as it does in many other sites. I can add/remove/change payment methods. That's it. 

In other words, I'm asking for payment methods to be manageable by the user as Azure/AWS/Netflix/..../.... etc does. 

I understand that there will have to be something kept on your side for subscriptions to work, I'm not opposing to that. I just want to have control over that to the same extend that I have with every other service I mentioned above. That was my original request, hopefully made a bit clearer now.

**Comment by mattdeluco** [comment]
@bs-git - could you please help me clarify your concerns? You mention copilot which is a Github product/service.

**Comment by bs-git** [comment]
@mattdeluco sorry for the confusion, I reported the issue to the wrong project - came directly from google and did not carefully read the project name - but you are not alone, even GitHub itself makes it hard to remove a payment method ;)

**Comment by fariskh** [comment]
Any solutions for this issue?

**Comment by wellloy1** [comment]
This is not legal! Why they still didn't add this feature? Everyone should be able to delete his private information. I cannot explain it...

**Comment by hanswang73** [comment]
> 

+1

**Comment by lostlang** [comment]
+1

**Comment by likanzhan** [comment]
+1

**Comment by andrii-yarmola** [comment]
+

**Comment by SergiuMituKTN** [comment]
+1

**Comment by mattdeluco** [comment]
Hi everyone,

I'm sorry to keep you all waiting. Before I can present to you the short term solution we have some work to do internally so that it works as expected.

I expect this solution to be available to you all by March 13.

**Comment by fariskh** [comment]
Hai mattdeluco, 
Thanks  a lot for this update.

**Comment by hanswang73** [comment]
Hai mattdeluco,
Thank you for your reply.

**Comment by hmehta89** [comment]
Thanks Matt

**Comment by mattdeluco** [comment]
Hello everyone,

I have an interim solution regarding deleting payment methods.

Right now it is possible to remove **non-default** payment methods in Docker Hub by navigating to Billing Details and looking under Payment Methods (for personal accounts and organizations.) Refer to the screenshots below: hover over any non-default card representing a payment method and click the "X" to remove it. This will remove your payment method and delete it from our payment gateway.

To delete your **default** payment method you'll need to send an email to `billing@docker.com` or `support@docker.com` requesting to have the method deleted. To expedite the process, please provide the expiry date and last four digits of the payment method. Note that the name from which we receive the request must match the name on the card. When our support team fulfills the request, as above, this will remove your payment method and delete it from our payment gateway.

We understand it would be preferable for this to be a self-service option from start to finish, including for default payment methods. We have scoped out a number of improvements for our payment flow that we intend to address in the future, including toggling auto-renewal.

I'm sorry you've waited so long, and thank you all for your patience.

Matt

<img width="835" alt="Screenshot 2023-03-10 at 10 38 35 AM" src="https://user-images.githubusercontent.com/2262420/224362293-d19c91d3-b54d-4eb8-bf44-8c674cf5ff66.png">

<img width="829" alt="Screenshot 2023-03-10 at 10 41 14 AM" src="https://user-images.githubusercontent.com/2262420/224362338-79bbe3ca-1223-4167-81ce-eb93f11c9285.png">

<img width="823" alt="Screenshot 2023-03-10 at 10 44 19 AM" src="https://user-images.githubusercontent.com/2262420/224362369-b9db1b2d-133d-4dc2-bbe3-94e0f6fd51cb.png">

**Comment by ReportOneMDU** [comment]
+1

**Comment by anarchy-79** [comment]
+1
------- Original Message -------

**Comment by anarchy-79** [comment]
I literally cancelled my credit card. Not joking. It was the fastest way. I get new ones in two days tops, same code, just update it once online and I'm golden. Not going to see my money any time soon, or ever again, that's for sure.
------- Original Message -------

**Comment by adamorlowskipoland** [comment]
+1

**Comment by uzair9** [comment]
I don't have a recurring subscription. Still, GitHub is not allowing me to take down my card. This is complete nonsense!

**Comment by neelima-loyyo** [comment]
Is this feature in? I signed up for a code pilot trial and have canceled my trial. I want to remove my payment information. How do I go about that?

**Comment by mattdeluco** [comment]
There seem to be some comments on this thread related to Github products and services. This thread is specifically about subscriptions on [Docker Hub](https://hub.docker.com).

**Comment by ashleysommer** [comment]
@uzair9990 This issue thread is about Dockerhub, Github is a different organization.
@neelima-loyyo Code Pilot is a github product. This issue thread is about Dockerhub.

**Comment by anarchy-79** [comment]
"That is not my department, see other department for further inquiries"
------- Original Message -------

**Comment by ashleysommer** [comment]
>"That is not my department, see other department for further inquiries"

Are you making a joke, or do people honestly think Github and Dockerhub are two departments of some secret megacorp?

**Comment by anarchy-79** [comment]
I am joking. I also acknowledge as a 3rd line operator your instinctive shirking of the issue.
------- Original Message -------

**Comment by mft25** [comment]
+1 for self-service removal of default card details

**Comment by ankudinov** [comment]
@mattdeluco Being one of the first users who reported this issue, I'm very surprised that it takes that long to resolve. Especially keeping in mind the community interest and obvious legal reasons. After all it's just a matter of introducing additional payment methods, like PayPal where nothing is exposed directly. TBH, I was never facing this issue with any other services I'm paying for. Nevertheless, please don't take it as hard critic. I totally understand possible challenges and have seen the message about the workaround. At least based on this thread I'm not loosing a hope that it will be resolved at some point.
May be if you could share some plans and timelines that would make community a bit happier?

**Comment by anarchy-79** [comment]
Yeah gee, I wonder why that is! It's almost as if they're another predatory capitalist corporation that cares about nothing but their bottom line and to suck people dry like the ticks they are..

I just cancelled my card and got a new one free of charge. It's basically all you can do at this point. Oh, also never EVER buy anything from these people ever again. Find out who they are, what other pots they have their tentacles in, and avoid them as the plague until the day you die. Nothing else to do. The system has got their backs.
------- Original Message -------

**Comment by sant123** [comment]
+1. We are in 2024 and still no way to remove the credit card 😕

**Comment by Artur-at-work** [comment]
+1 We are in 2024 and still no way to remove the credit card

**Comment by bliksund-khiem** [comment]
+1 We are in 2024 and still no way to remove the credit card

**Comment by pritampawade** [comment]
+1, just why you are not letting us delete our payment information? Never expected such a cheap practice from GitHub.

**Comment by Olivier-Agm** [comment]
This is just crazy. I just subscribed to GITHUB for the company I'm working for, requesting the company credit card to perform the yearly payment. When entered, it says "Save payment information" (or something similar) so I cannot really say I was not aware the card info will not be saved. I told my boss about it and, as I expected, he did not agree that the card info should be saved somewhere on the cloud. Of course it was handy to subscribe furthermore to Copilot, but now how the hell do I remove the card info?!? We'll need it only next year...

**Comment by ashleysommer** [comment]
@pritampawade  @Olivier-Agm 

Wrong thread. This is an issue about Dockerhub's shady practices.
I acknowledge they are both guilty of the same anti-consumer behaviour, but this is like calling McDonald's customer support line to complain about Burger King's fries. Please move your complaints about Github to the correct Github complaints thread.

**Comment by ashleysommer** [comment]
This is still not resolved. Keep it open.

**Comment by fariskh** [comment]
Still not resolved??
